### PR TITLE
Add cron time parameters

### PR DESCRIPTION
**Pull request by matonb**
Accepts puppet cron type time attributes

#### Pull Request (PR) description
Enable user to define cron time attributes instead of defaulting to random time

#### This Pull Request (PR) fixes the following issues
#47 It doesn't fix this issue per-se but does allow users to specify more than one execution time

### CODE REVIEW

File: manifests/certonly.pp [hunk 1]
```diff
@@ -31,20 +31,30 @@
...
+  Enum['apache', 'standalone', 'webroot', 'nginx'] $plugin = 'standalone',
+  Optional[Array] $webroot_paths                           = undef,
+  String $letsencrypt_command                              = $letsencrypt::command,
+  Optional[Array] $additional_args                         = undef,
+  Array $environment                                       = [],
+  Boolean $manage_cron                                     = false,
+  Boolean $suppress_cron_output                            = false,
+  $cron_before_command                                     = undef,
+  $cron_success_command                                    = undef,
+  # 0 - 23, seed is title plus fqdn
```
Comment (ekohl) [inline manifests/certonly.pp]: I think this comment can be removed since the parameter is documented

File: manifests/certonly.pp [hunk 2]
```diff
@@ -31,20 +31,30 @@
...
+  Variant[Integer, String, Array] $cron_hour               = fqdn_rand(24, $title),
```
Comment (ekohl) [inline manifests/certonly.pp]: You can limit it further by specifying `Integer[0, 23]`. At least, I think that was inclusive. Also wondering if the `Array` should be further specified. A similar comment applies to the minute parameter.

**Comment by bastelfreak** [comment]
Hi @matonb, could you rebase please?

**Comment by matonb** [comment]
@bastelfreak rebased

File: manifests/certonly.pp [hunk 3]
```diff
@@ -125,3 +131,4 @@
     }
   }
 }
+
```
Comment (ekohl) [inline manifests/certonly.pp]: Nit: could you remove this?

Comment (matonb) [reply manifests/certonly.pp]: @ekohl Done

**Comment by alexjfisher** [comment]
@matonb Thanks for the squash/rebase. Would you like to amend the final commit message though?

File: README.md [hunk 4]
```diff
@@ -129,17 +129,21 @@ letsencrypt::certonly { 'foo':
 }
 ```
 
-To automatically renew a certificate, you can pass the `manage_cron` parameter.
-You can optionally add a shell command to be run on success using the `cron_success_command` parameter.
-You can optionally add a shell command to be run on before using the `cron_before_command` parameter.
-You can optionally specify one or multiple days of the month when to run the cron using the `cron_monthday` parameter (default is every day).
+To automatically renew a certificate, you can pass the `manage_cron` parameter.
```
Comment (ekohl) [inline README.md]: Could you remove the trailing whitespace here?

Comment (matonb) [reply README.md]: Added to for newlines when renderd

Comment (ekohl) [reply README.md]: A lot of scripts and editors automatically strip trailing whitespace and it's also very hard to review. It's probably better to rewrite as a list:
* `manage_cron` can be used to automatically renew the certificate
* `cron_success_command` can be used to run a shell command on a successful renewal
* `cron_before_command` can be used to run a shell command before a renewal
* ....

**Comment by alexjfisher** [comment]
@matonb Thanks!  Sorry for the delay in getting this in.

**Comment by Dan33l** [comment]
@matonb Thank you.
Your PR is now release on the puppet forge.

**Pull request merged**